File: algo_4_erfen/L540-m.cpp
//
// Created by 李恒 on 2020/6/28.
//
#include <iostream>
#include <vector>

using namespace std ;
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int lt=0, rt=nums.size()-1, mid;
        while(lt < rt){
            mid = lt + (rt-lt)/2;
            if(mid%2 == 0){
                if(nums[mid] == nums[mid-1])
                    rt = mid-2;
                else
                    lt = mid;
            }
            else{
                if(nums[mid] == nums[mid-1])
                    lt = mid+1;
                else
                    rt = mid-1;
            }
        }
        return nums[lt];
    }
};

File: python-algorithm/algo_04_erfen/L540-m.py
#--coding:utf-8--


class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        lt, rt = 0, len(nums)-1
        while lt < rt:
            mid = lt + (rt-lt)//2
            if mid%2 == 0:
                if nums[mid] == nums[mid-1]:
                    rt = mid-2
                else:
                    lt = mid
            else:
                if nums[mid] == nums[mid-1]:
                    lt = mid + 1
                else:
                    rt = mid - 1
        return nums[lt]